pallets/msa/src/lib.rs

```diff
@@ -556,6 +571,51 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Retire a MSA
+		///
+		/// ### Events
+		/// - Deposits [`MsaRetired`](Event::MsaRetired) when MSA is retired
+		///
+		/// ### Errors
+		///
+		/// - Returns [`NoKeyExists`](Error::NoKeyExists) if `delegator` does not have an MSA key.
+		/// - Returns [`MoreThanOneKeyExists`](Error::MoreThanOneKeyExists) if the MSA has more than one account key.
+		/// - Returns [`RegisteredProviderCannotBeRetired`](Error::RegisteredProviderCannotBeRetired) if the MSA id is a registered provider
+
+		#[pallet::weight((T::WeightInfo::retire_msa(), DispatchClass::Normal, Pays::Yes))]
+		pub fn retire_msa(origin: OriginFor<T>) -> DispatchResult {
+			// Check and get the account id from the origin
```

Comment:
this comment is not completely accurate. I can see why this is tricky. This actually represents an account key in our system and not an account id.